lua/rocks-dev/local-rock-handler.lua

```diff
@@ -14,7 +14,7 @@ local rock_handler = {}
 ---@param rock RockSpec
 ---@return async fun(report_progress: fun(message: string), report_error: fun(message: string)) | nil
 function rock_handler.get_sync_callback(rock)
-    if rock.dir then
+    if rock.dir or rock.dev then
```

Comment:
suggestion: If `rock.dir` is not set, but `rock.dev` is, we should probably also make sure `user_configuration.dev.path` is set.